archive/archive.go

```diff
@@ -26,7 +26,38 @@ func ArchiveDirectory(dir, outputFile string) error {
 		return fmt.Errorf("failed to create archive: %w", err)
 	}
 	if err := os.Rename(tempTarArchive, outputFile); err != nil {
-		return fmt.Errorf("failed to rename temporary tar bundle to output file: %w", err)
+		return fmt.Errorf("failed to rename temporary archive to output file: %w", err)
 	}
 	return nil
 }
+
+func UnarchiveToDirectory(archive, destDir string) error {
+	tarArchive, err := os.Open(archive)
+	if err != nil {
+		return fmt.Errorf("failed to open archive for extraction: %w", err)
+	}
+	defer tarArchive.Close()
+
+	archiverByExtension, err := archiver.ByExtension(archive)
+	if err != nil {
+		return fmt.Errorf("failed to identify archive format: %w", err)
+	}
+
+	unarc, ok := archiverByExtension.(archiver.Unarchiver)
+	if !ok {
+		return fmt.Errorf("not an valid archive extension")
+	}
+
+	switch t := unarc.(type) {
+	case *archiver.TarGz:
+		t.OverwriteExisting = true
+	case *archiver.Tar:
+		t.OverwriteExisting = true
+	}
```

Comment:
Are these 2 cases supposed to be the same? It could be simplified like:
```suggestion
	switch t := unarc.(type) {
	case *archiver.TarGz, case *archiver.Tar:
		t.OverwriteExisting = true
	}
```

Reply:
I assume you mean

```
    switch t := unarc.(type) {
    case *archiver.TarGz, *archiver.Tar:
        t.OverwriteExisting = true
    }
```

but this doesn't work - see https://go.dev/ref/spec#Type_switches:

> The TypeSwitchGuard may include a short variable declaration. When that form is used, the variable is declared at the end of the TypeSwitchCase in the implicit block of each clause. In clauses with a case listing exactly one type, the variable has that type; otherwise, the variable has the type of the expression in the TypeSwitchGuard.